crates/sui-types/src/temporary_store.rs

```diff
@@ -273,20 +288,20 @@ impl<S> TemporaryStore<S> {
                     package_id: ctx.package_id,
                     transaction_module: ctx.transaction_module.clone(),
                     sender: ctx.sender,
-                    object_id: id,
+                    object_id: *id,
                     version,
                 },
             };
             events.push(event);
-            deleted.insert(id, (version, kind));
+            deleted.insert(*id, (version, *kind));
         }
 
         // Combine object events with move events.
-        events.extend(self.events);
+        events.extend_from_slice(&self.events);
 
         InnerTemporaryStore {
-            objects: self.input_objects,
-            mutable_inputs: self.mutable_input_refs,
+            objects: self.input_objects.clone(),
+            mutable_inputs: self.mutable_input_refs.clone(),
```

Comment:
Would really like to avoid these

Reply:
I'm not sure what lifetimes have to do with it since you are no longer taking the struct by-value. Can you point to the area where you needed to borrow this? I'm having trouble following